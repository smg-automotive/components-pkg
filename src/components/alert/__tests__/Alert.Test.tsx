import React from 'react';

import { render, screen } from '@testing-library/react';

import Alert from '../index';

const MockIcon = () => <svg data-testid="test-icon" />;

describe('<Alert>', () => {
  it('renders Alert with title', () => {
    render(
      <Alert
        title="I am the title"
        typeMessage="info"
        description="I am a description"
        icon={<MockIcon />}
      />
    );

    expect(screen.getByText('I am the title')).toBeInTheDocument();
  });

  it('renders Alert with link', () => {
    render(
      <Alert
        typeMessage="info"
        description="I am a description"
        link={{
          url: 'https://www.autoscout24.ch/de',
          text: 'Link',
        }}
        icon={<MockIcon />}
      />
    );

    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('href', 'https://www.autoscout24.ch/de');
  });

  describe('Variations', () => {
    it('renders information alert message correctly', () => {
      const view = render(
        <Alert
          typeMessage="info"
          description="I am a description"
          icon={<MockIcon />}
        />
      );

      expect(view.container).toMatchSnapshot();
    });

    it('renders error alert message correctly', () => {
      const view = render(
        <Alert
          typeMessage="error"
          description="I am a description"
          icon={<MockIcon />}
        />
      );

      expect(view.container).toMatchSnapshot();
    });

    it('renders warning alert message correctly', () => {
      const view = render(
        <Alert
          typeMessage="warning"
          description="I am a description"
          icon={<MockIcon />}
        />
      );

      expect(view.container).toMatchSnapshot();
    });

    it('renders success alert message correctly', () => {
      const view = render(
        <Alert
          typeMessage="success"
          description="I am a description"
          icon={<MockIcon />}
        />
      );

      expect(view.container).toMatchSnapshot();
    });
  });
});
