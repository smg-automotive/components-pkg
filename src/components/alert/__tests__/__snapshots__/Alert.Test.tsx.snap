// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert> Variations renders error alert message correctly 1`] = `
<div>
  <div
    class="chakra-alert css-0"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-0"
    >
      <svg
        data-testid="test-icon"
      />
    </span>
    <div
      class="css-0"
    >
      <div
        class="chakra-alert__desc css-0"
      >
        I am a description
      </div>
    </div>
  </div>
</div>
`;

exports[`<Alert> Variations renders information alert message correctly 1`] = `
<div>
  <div
    class="chakra-alert css-0"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-0"
    >
      <svg
        data-testid="test-icon"
      />
    </span>
    <div
      class="css-0"
    >
      <div
        class="chakra-alert__desc css-0"
      >
        I am a description
      </div>
    </div>
  </div>
</div>
`;

exports[`<Alert> Variations renders success alert message correctly 1`] = `
<div>
  <div
    class="chakra-alert css-0"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-0"
    >
      <svg
        data-testid="test-icon"
      />
    </span>
    <div
      class="css-0"
    >
      <div
        class="chakra-alert__desc css-0"
      >
        I am a description
      </div>
    </div>
  </div>
</div>
`;

exports[`<Alert> Variations renders warning alert message correctly 1`] = `
<div>
  <div
    class="chakra-alert css-0"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-0"
    >
      <svg
        data-testid="test-icon"
      />
    </span>
    <div
      class="css-0"
    >
      <div
        class="chakra-alert__desc css-0"
      >
        I am a description
      </div>
    </div>
  </div>
</div>
`;
